cdk/backend.ts

```diff
@@ -163,7 +163,22 @@ const coapDockerfilePath = path.join(
 	'containers',
 	'coap',
 )
-const coapHash = await hashFolder(coapDockerfilePath)
+const coapSimulatorDownloadUrl = (
+	await getCoAPHealthCheckSettings({
+		ssm,
+		stackName: STACK_NAME,
+	})
+).simulatorDownloadURL.toString()
+const { headers, ok, status } = await fetch(coapSimulatorDownloadUrl, {
+	method: 'HEAD',
+})
+if (!ok) throw new Error(`Failed to download CoAP simulator: ${status}`)
+
+const coapSimulatorBinaryHash = `${headers.get('etag')}-${headers.get(
+	'content-length',
+)}-${headers.get('last-modified')}`
```

Comment:
Instead of using the server data, use the checksum of the file. Because that is what really matters and will tell if the contents have changed, not the metadata the webserver has.

Reply:
I did this at the first time, but we need to move the downloaded file to `cdk/resources/containers/coap` where it is docker context. Otherwise, docker won't allow you to copy the files to docker image. That's why I re-think and use server meta data to check.

Reply:
You can download it to that location, no problem.

Reply:
This solution depends then on S3, which is OK, but needs to be documented. Because not every file host automatically provides etags.